src/elements/ngm-nav-tools.ts

```diff
@@ -159,21 +170,35 @@ export class NgmNavTools extends LitElementI18n {
   onLeftDown(event) {
     const pickedObject = this.viewer!.scene.pick(event.position);
     if (pickedObject && pickedObject.id && pickedObject.id.id === this.refIcon.id) {
-      this.viewer!.scene.screenSpaceCameraController.enableInputs = false;
-      this.eventHandler!.setInputAction(event => this.onMouseMove(event), ScreenSpaceEventType.MOUSE_MOVE);
-      this.viewer!.scene.camera.lookAtTransform(Matrix4.IDENTITY);
-      this.viewer!.trackedEntity = undefined;
+      this.stopTracking();
       this.moveRef = true;
     }
   }
 
   onLeftUp() {
     if (!this.moveRef) return;
     this.moveRef = false;
-    this.viewer!.trackedEntity = this.refIcon;
+    this.startTracking();
+  }
+
+  stopTracking() {
+    this.viewer!.scene.screenSpaceCameraController.enableInputs = false;
+    this.eventHandler!.setInputAction(event => this.onMouseMove(event), ScreenSpaceEventType.MOUSE_MOVE);
+    this.viewer!.scene.camera.lookAtTransform(Matrix4.IDENTITY);
+  }
+
+  startTracking() {
+    this.addTargetPoint(this.refIcon.position!.getValue(this.julianDate));
+    const center = this.refIcon.position!.getValue(this.julianDate);
+    const camera = this.viewer!.camera;
+    lookAtPoint(center, camera);
+    const transform = Transforms.eastNorthUpToFixedFrame(center);
+    camera.lookAtTransform(transform);
```

Comment:
Why do a lookAtPoint first then a lookAtTransform? Is the first line useful?

Reply:
lookAtTransform start tracking of position